scripts/globals/pasteventwatcher.lua

```diff
@@ -0,0 +1,220 @@
+-----------------------------------
+-- Past Event Watcher NPCs
+-----------------------------------
+
+-----------------------------------
+
+require("scripts/globals/quests")
+require("scripts/globals/missions")
+require("scripts/globals/zone")
+
+pasteventwatcher = {}
+
+local gobCS = -- add the goblin cs to this table
+{
+    [102] = 126, -- La Theine Plateau
+    [106] = 247, -- North Gustaberg
+    [107] = 905, -- South Gustaberg
+    [119] = 35,  -- Meriphataud Mountains
+    [122] = 6,   -- Ro'Maeve
+    [162] = 10,  -- Castle Zvahl Keep
+    [167] = 100, -- Bostaunieux Oubliette
+    [198] = 67,  -- Maze of Shakhrami
+    [200] = 61,  -- Garlaige Citadel
+}
+
+function pasteventwatcher.rewatch(player, trigger)
+    local zone = player:getZoneID()
+    local ID = zones[zone]
+    local options = 1
+    local cs = -- add checks to this table
+    {
+        [102] = -- La Theine Plateau
+        {
+            [1] = function() return (player:hasCompletedQuest(BASTOK, dsp.quest.id.bastok.DARK_PUPPET)) end,
+            [2] = function() return (player:hasCompletedQuest(WINDURST, dsp.quest.id.windurst.HITTING_THE_MARQUISATE)) end,
+            [3] = function() return (player:hasCompletedQuest(WINDURST, dsp.quest.id.windurst.I_CAN_HEAR_A_RAINBOW)) end,
+            [4] = function() return (player:hasCompletedQuest(SANDORIA, dsp.quest.id.sandoria.A_TIMELY_VISIT)) end,
+            [5] = function() return (player:hasCompletedQuest(SANDORIA, dsp.quest.id.sandoria.A_TIMELY_VISIT)) end,
+            [6] = function() return (player:hasCompletedMission(COP, dsp.mission.id.cop.THE_MOTHERCRYSTALS)) end,
+            [7] = function() return (player:hasCompletedMission(COP, dsp.mission.id.cop.THREE_PATHS)) end,
+            [8] = function() return (player:completeQuest(OTHER_AREAS_LOG, dsp.quest.id.otherAreas.WAKING_THE_BEAST)) end,
+            [9] = function() return (player:completeQuest(OTHER_AREAS_LOG, dsp.quest.id.otherAreas.WAKING_THE_BEAST)) end,
+            [10] = function() return (player:hasCompletedQuest(JEUNO, dsp.quest.id.jeuno.CHOCOBO_ON_THE_LOOSE)) end,
+            [11] = function() return (player:hasCompletedQuest(ABYSSEA, dsp.quest.id.abyssea.A_GOLDSTRUCK_GIGAS)) end,
+            [12] = function() return (player:hasCompletedQuest(ABYSSEA, dsp.quest.id.abyssea.A_GOLDSTRUCK_GIGAS)) end,
+            [13] = function() return (player:hasCompletedQuest(ABYSSEA, dsp.quest.id.abyssea.AN_OFFICER_AND_A_PIRATE)) end,
+            [14] = function() return (player:hasCompletedQuest(ABYSSEA, dsp.quest.id.abyssea.TENUOUS_EXISTENCE)) end,
+            [15] = false,               -- QUEST NOT IMPLEMENTED (ROV)
```

Comment:
If you actually have all these bits figured out, go ahead and add them. 

Don't worry about the quests not currently working, your script will work once they do.

Reply:
@zynjec  don't have the correct CS ID for those ones. left them as false and with a note once those quest are implemented they can be added. (CS for ROV are weird)